src/rpc/connectivity/cluster.cc

```diff
@@ -1075,26 +1105,29 @@ void connectivity_cluster_t::run_t::handle(
         logERR("Connected to peer with unresolvable hostname%s: %s, closing "
                "connection.  Consider using the '--canonical-address' launch option.",
                other_peer_addr_hosts.size() > 1 ? "s" : "", hostnames.c_str());
-        return;
+        return join_result_t::TEMPORARY_ERROR;
     }
 
     /* Sanity checks */
     if (other_id == parent->me) {
         // TODO: report this on command-line in some cases. see issue 546 on github.
-        return;
+        return join_result_t::PERMANENT_ERROR;
     }
+
     if (other_id.is_nil()) {
         logERR("Received nil peer id from %s, closing connection.", peername);
-        return;
+        return join_result_t::PERMANENT_ERROR;
```

Comment:
I'm not sure in what cases this can happen to be honest (I've never seen it happen either).
However if in doubt, we should return a TEMPORARY_ERROR.